src/Lender.sol

```diff
@@ -34,12 +34,9 @@ contract Lender {
     /// @notice flag that allows admin to stop all lending and minting across the entire protocol
     bool public halted;
 
-    /// @notice contract used to execute swaps on Swivel's exchange
-    address public immutable swivelAddr;
-    /// @notice a SushiSwap router used by Pendle to execute swaps
-    address public immutable pendleAddr;
-    /// @notice a pool router used by APWine to execute swaps
-    address public immutable apwineAddr;
+    /// @notice protocol specific addresses that adapters reference when executing lends
+    /// @dev these addresses are references by an implied enum; adapters hardcode the index for their protocol
+    address[] public protocolAddressess;
```

Comment:
Hmm, I actually am not 100% on the way solidity handles unbounded arrays, if we lets say have something at index 0 and 1, then add something to index 5, what is returned for array.length() ?

Is it 6 and in this case address(0) is generated for all the other values up to index 5?

Reply:
With dynamically sized arrays, we can only edit existing elements in the array (in this case 0 and 1) or add to the array by calling `push`. For array.length, it returns the number of elements that have been created for it. So if you create an empty array (length 0) and call `push` three times, the length would be 3. 

> Is it 6 and in this case address(0) is generated for all the other values up to index 5?

No - such an operation would not be possible. If you really wanted to add to the 5th index, you'd have to call `push(address(0))` 4 times before pushing onto the 5th index.